### PR TITLE
PP-9407 add new endpoint and implement validation

### PR DESCRIPTION
**Pull request by SandorArpa**
## WHAT YOU DID
- add a new end point for authorisation API request
- implement validation of request payload
- add/update tests

### CODE REVIEW

File: src/test/java/uk/gov/pay/api/json/AuthorisationAPIRequestDeserializerTest.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        assertThat(request.getOneTimeToken(), is("27432234tyu-4567"));
+        assertThat(request.getCardNumber(), is("12345678901234"));
+        assertThat(request.getCvc(), is("123"));
+        assertThat(request.getExpiryDate(), is("09/28"));
+        assertThat(request.getCardholderName(), is("Joe Boggs"));
+    }
+
+    @Test
+    void deserialize_shouldThrowBadAuthorisationAPIRequestException_AsExpiryDateMissing_whenExpiryDateIsNullValue() throws Exception {
+        // language=JSON
```
Comment (kbottla) [inline src/test/java/uk/gov/pay/api/json/AuthorisationAPIRequestDeserializerTest.java:50]: Remove 
```suggestion
        // language=JSON
```

File: src/test/java/uk/gov/pay/api/json/AuthorisationAPIRequestDeserializerTest.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+import uk.gov.pay.api.exception.BadAuthorisationAPIRequestException;
+import uk.gov.pay.api.model.AuthorisationAPIRequest;
+import uk.gov.pay.api.model.PaymentError;
+
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThrows;
+
+class AuthorisationAPIRequestDeserializerTest {
```
Comment (kbottla) [inline src/test/java/uk/gov/pay/api/json/AuthorisationAPIRequestDeserializerTest.java:18]: Would be good to parameterise these tests so different criteria (blank, missing, length, type) can be tested without repeating similar setup/assertions across.

File: src/main/java/uk/gov/pay/api/model/AuthorisationAPIRequest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
+package uk.gov.pay.api.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class AuthorisationAPIRequest {
```
[the remark this answers is not in the document]
Comment (SandorArpa) [reply src/main/java/uk/gov/pay/api/model/AuthorisationAPIRequest.java:5]: 👍

File: src/main/java/uk/gov/pay/api/exception/BadAuthorisationAPIRequestException.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+public class BadAuthorisationAPIRequestException extends RuntimeException {
+
+    private PaymentError paymentError;
+
+    public BadAuthorisationAPIRequestException(PaymentError paymentError) {
+        this.paymentError = paymentError;
+    }
+
+    public PaymentError getPaymentError() {
```
[the remark this answers is not in the document]
Comment (SandorArpa) [reply src/main/java/uk/gov/pay/api/exception/BadAuthorisationAPIRequestException.java:13]: 👍

**Pull request closed without merging**